src/core/core.controller.js

```diff
@@ -31,10 +31,74 @@ module.exports = function(Chart) {
 		data.datasets = data.datasets || [];
 		data.labels = data.labels || [];
 
-		config.options = helpers.configMerge(
+
+		// Merge options for mixed charts
+		// [1] Does not work if axes are stacked
+		// [2] Only triggered if there are at least 2 different types
```

Comment:
it seems like the code would be simpler if you always did it. is this required?

Reply:
If I just replace `if (isMixed) {` with `if (true/*isMixed*/) {`, `gulp test` gives me the following : 
```
Chrome 63.0.3239 (Linux 0.0.0): Executed 683 of 687 (130 FAILED) (skipped 4) ERROR (8.176 secs / 7.324 secs)
Firefox 52.0.0 (Linux 0.0.0): Executed 683 of 687 (130 FAILED) (skipped 4) (14.328 secs / 12.872 secs)
```
:/

And for stacked axes, without the check, 2 tests are failing.

Reply:
Fixed it ! 
```
if (dataset.type !== undefined) {
	config.options = helpers.configMerge(defaults[dataset.type], config.options || {});
}
```